# Statistical Analysis

## Introduction

This notebook applies statistical methods to investigate relationships between key variables in the Saudi job market dataset.

The goal is not only to describe the data but also to determine whether observed patterns are statistically significant.

The analysis includes:
- Chi-Square Test of Independence
- Statistical interpretation
- Business implications

In [26]:
import pandas as pd
import numpy as np

from scipy.stats import chi2_contingency, f_oneway

## Statistical Question 1

### Is there a statistically significant association between industry and gender requirements in job postings?

### Hypotheses

**Null Hypothesis (H₀):**
There is no association between industry and gender requirements.

**Alternative Hypothesis (H₁):**
There is a significant association between industry and gender requirements.

In [27]:
df = pd.read_csv("data/processed/Jadarat_cleaned.csv")

df.head()

,job_title,job_date,job_desc,job_tasks,comp_name,comp_no,comp_type,comp_size,eco_activity,qualif,region,city,benefits,contract,positions,job_post_id,exper,gender
0,محاسب,27/05/1444,['إعداد وتنظيم مستندات الصرف ومتابعة تحصيل الإ...,[' تدقيق المطالبات المالية والتأكد من اكتمال...,شركة مقر العالم للسفريات,1-317262,خاص,متوسطة فئة أ,أنشطة وكالات السياحة والسفر,"['Language data', 'اللغة الانجليزية', 'محترف']",الرياض,AR RIYADH...,"['Salary', '5000.0']",دوام كامل,0 / 1,20202026350419,0 Years,both
1,بائع,27/05/1444,['بيع مجموعة من السلع والخدمات للعملاء، وتوفير...,"[' بيع مجموعة من السلع والخدمات للعملاء.', '...",شركة عالم الكهرباء للمقاولات,4-1324428,خاص,متوسطة فئة ب,تركيب انظمة التبريد وتكييف الهواء وصيانتها واص...,NaN,المنطقة الشرقية,AD DAMMAM...,"['Salary', '5000.0']",دوام كامل,0 / 3,20202026350389,0 Years,both
2,أخصائي عمليات موارد بشرية,27/05/1444,['تنفيذ الإجراءات والأنظمة والنماذج الخاصة بمر...,[' تنفيذ الإجراءات والأنظمة والنماذج الخاصة ...,شركه دار السلام,1-155294,خاص,متوسطة فئة أ,ترميمات المباني السكنية والغير سكنية,"['Language data', 'الانجليزيه', 'محترف']",الرياض,ATH THUMA...,"['Salary', '4000.0']",دوام كامل,0 / 2,20202026350347,2 Years,both
3,ميكانيكي سيارات,27/05/1444,['تشخيص أعطال السيارات وإصلاحها وتنفيذ برامج ا...,[' فحص أداء المعدّات الكهربائية والميكانيكية...,مؤسسة لمكو لغيار الزيوت,8-1925495,خاص,صغيرة فئة ب,NaN,"['Skill data', 'صيانة السيارات وتقييم الاعطال'...",المنطقة الشرقية,AL HUFUF...,"['Salary', '5000.0']",دوام كامل,0 / 10,20202026350219,0 Years,M
4,محاسب,27/05/1444,['إعداد وتنظيم مستندات الصرف ومتابعة تحصيل الإ...,[' تدقيق المطالبات المالية والتأكد من اكتمال...,مؤسسة فكرة اليمامة للمقاولات,1-2356639,خاص,كبيرة,الإنشاءات العامة للمباني السكنية,"['Skill data', 'تحمل ضغط العمل', 'محترف', 'Lan...",الرياض,AR RIYADH...,"['Salary', '5000.0']",دوام كامل,0 / 1,20202026350043,0 Years,both


In [28]:
contingency_table = pd.crosstab(
    df["eco_activity"],
    df["gender"]
)

contingency_table.head()

gender,F,M,both
eco_activity,,,
تجارة الجملة والتجزئة - تكنولوجيا، والآلات والملابس والخدمات وغيرها,0,0,1
مستشفيات الرعاية الطبية الممتدة والنقاهة,1,0,0
أصحاب المعدات وسائقي النقل (خاصة),0,0,1
أعمال وتركيب الديكورات المختلفة,0,0,2
أنشطة أخرى لأنشطة البيع بالجملة والتجزئة لقطع غيار السيارات وتوابعها ، يشمل ( تنجيد السيارات ),0,1,0


In [29]:
chi2, p_value, dof, expected = chi2_contingency(contingency_table)

print(f"Chi-Square Statistic: {chi2:.4f}")
print(f"P-value: {p_value:.4f}")
print(f"Degrees of Freedom: {dof}")

Chi-Square Statistic: 1210.2433
P-value: 0.0000
Degrees of Freedom: 720


In [44]:


n = contingency_table.to_numpy().sum()

cramers_v = np.sqrt(chi2 / (n * (min(contingency_table.shape) - 1)))

print(f"Cramér's V: {cramers_v:.4f}")

Cramér's V: 0.2353


### Interpretation

A Chi-Square Test of Independence was conducted to examine the relationship between industry and gender requirements in job postings.

The results showed a statistically significant association between the two variables (χ² = 1210.24, p < 0.001). Therefore, the null hypothesis was rejected.

To assess the strength of this relationship, Cramér's V was calculated. The result (V = 0.6622) indicates a **strong association** between industry and gender requirements.

This suggests that gender preferences in job advertisements vary substantially across different industries.

### Business Insight

The analysis indicates that hiring preferences are strongly associated with industry type.

Some industries predominantly advertise positions for male applicants, while others recruit female applicants or accept both genders. These differences are statistically significant and represent meaningful hiring patterns rather than random variation.

Understanding these patterns can help policymakers monitor labor market trends and assist organizations in evaluating recruitment practices.

## Statistical Question 2

### Is there a statistically significant association between region and industry in Saudi Arabia?

In [32]:
top20_industries = df["eco_activity"].value_counts().head(20).index 

df["industry_grouped"] = df["eco_activity"].apply(
    lambda x: x if x in top20_industries else "Other"
)

In [33]:
df["industry_grouped"].value_counts()

industry_grouped
Other                                                                                                                            1006
المطاعم مع الخدمة                                                                                                                  65
الخدمات الطبية بالمصحات المتخصصة ومراكز إعادة التأهيل والصحة الوقائية                                                              46
أنشطة خدمات صيانة المباني                                                                                                          44
أنشطة الاستشارات الهندسية والمعمارية                                                                                               27
ترميمات المباني السكنية والغير سكنية                                                                                               24
الإنشاءات العامة للمباني السكنية                                                                                                   24
محلات الوجبات السريعة ( تشمل محلات البيتزاء )

In [34]:
contingency_table = pd.crosstab(
    df["region"],
    df["industry_grouped"]
)

contingency_table

industry_grouped,Other,أنشطة الاستشارات الهندسية والمعمارية,أنشطة المحاماة والاستشارات القانونية,أنشطة خدمات صيانة المباني,الإنشاءات العامة للمباني السكنية,الانشاءات العامة للمباني غير السكنية (مثل المدارس والمستشفيات والفنادق ....الخ),التسجيل لتقديم خدمات الحوسبة السحابية,التنظيف العام للمباني,الخدمات الطبية بالمصحات المتخصصة ومراكز إعادة التأهيل والصحة الوقائية,المجمعات الطبية العامة,...,النقل البري للبضائع,بيع الأغذية والمشروبات بالتجزئة في الأكشاك والأسواق,ترميمات المباني السكنية والغير سكنية,تشغيل المقاصف والكفتيريات على أساس إمتياز محدد ، يشمل ( المقاصف والكفتيريات في المصانع والمكاتب والمستشفيات والمدارس .. الخ ),تصميم وبرمجة البرمجيات الخاصة,تقديم خدمات تسويقية نيابةً عن الغير,تلميع السيارات (البوليش),شراء وبيع الاراضى والعقارات وتقسيمها وأنشطة البيع على الخارطة,محلات الوجبات السريعة ( تشمل محلات البيتزاء ),محلات تقديم المشروبات ( الكوفي شوب )
region,,,,,,,,,,,,,,,,,,,,,
اخرى,0,0,0,0,0,0,0,0,0,0,...,0,0,1,0,0,0,0,0,0,0
الباحة,16,0,0,8,0,0,0,0,9,2,...,0,2,5,0,0,0,0,0,0,1
الجوف,5,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
الحدود الشمالية,3,1,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
الرياض,452,13,11,16,10,4,5,12,12,1,...,3,11,7,7,3,10,1,3,14,6
القصيم,24,0,0,0,0,1,0,0,3,0,...,1,1,0,0,0,0,0,0,0,2
المدينة المنورة,60,2,1,3,1,0,0,0,6,1,...,1,0,2,2,0,0,0,2,0,1
المنطقة الشرقية,145,1,3,6,3,3,11,0,4,1,...,0,0,1,6,2,1,8,2,0,2
تبوك,1,0,1,0,0,0,0,1,0,2,...,0,0,0,0,6,0,0,0,0,0


In [35]:
chi2, p_value, dof, expected = chi2_contingency(contingency_table)

print(f"Chi-Square Statistic: {chi2:.4f}")
print(f"P-value: {p_value:.4f}")
print(f"Degrees of Freedom: {dof}")

Chi-Square Statistic: 1058.1825
P-value: 0.0000
Degrees of Freedom: 260


In [36]:
n = contingency_table.to_numpy().sum()

cramers_v = np.sqrt(
    chi2 / (n * (min(contingency_table.shape) - 1))
)

print(f"Cramér's V: {cramers_v:.4f}")

Cramér's V: 0.2353


### Interpretation

A Chi-Square Test of Independence was conducted to examine the relationship between region and industry.

The results revealed a statistically significant association between the two variables (χ² = 1058.18, p < 0.001). Therefore, the null hypothesis was rejected.

Cramér's V was calculated to measure the strength of the association. The result (V = 0.2353) indicates a **weak association** between region and industry.

This suggests that although the distribution of industries varies across regions, the relationship is relatively weak.

### Business Insight

The analysis suggests that industry distribution differs across Saudi regions, but the differences are relatively modest.

This indicates that many industries are present in multiple regions rather than being concentrated in only a few locations. Organizations and policymakers can use this information to understand regional economic diversity and identify areas for future investment or workforce development.

## Statistical Question 3

### Business Question

Do different industries require different average years of experience?

In [37]:
df["exper"].value_counts().sort_index()

exper
0 Years     833
10 Years     15
12 Years      3
2 Years     436
4 Years     153
7 Years      30
Name: count, dtype: int64

In [45]:
df["exper_years"] = df["exper"].apply(
    lambda x: int(x.split()[0])
) 

In [46]:
df.to_csv(
    "data/processed/Jadarat_cleaned.csv",
    index=False
)

In [39]:
df[["exper", "exper_years"]].head(10)

,exper,exper_years
0,0 Years,0
1,0 Years,0
2,2 Years,2
3,0 Years,0
4,0 Years,0
5,0 Years,0
6,4 Years,4
7,4 Years,4
8,2 Years,2
9,4 Years,4


In [40]:
df.groupby("industry_grouped")["exper_years"].mean().sort_values(ascending=False)

industry_grouped
التسجيل لتقديم خدمات الحوسبة السحابية                                                                                            3.941176
أنشطة الاستشارات الهندسية والمعمارية                                                                                             3.444444
تصميم وبرمجة البرمجيات الخاصة                                                                                                    2.214286
شراء وبيع الاراضى والعقارات وتقسيمها وأنشطة البيع على الخارطة                                                                    1.894737
تقديم خدمات تسويقية نيابةً عن الغير                                                                                              1.333333
Other                                                                                                                            1.311133
أنشطة المحاماة والاستشارات القانونية                                                                                             1.300000
الإنشاءات العامة 

In [42]:
groups = [
    group["exper_years"].values
    for _, group in df.groupby("industry_grouped")
]

len(groups)

21

In [43]:
f_statistic, p_value = f_oneway(*groups)

print(f"F-statistic: {f_statistic:.4f}")
print(f"P-value: {p_value:.4f}")

F-statistic: 5.7882
P-value: 0.0000


### Interpretation

A one-way ANOVA was conducted to determine whether the average years of experience required differ across industries.

The results showed a statistically significant difference in average years of experience among industries (F = 5.79, p < 0.001).

Therefore, the null hypothesis was rejected, indicating that at least one industry requires a significantly different average level of experience compared to the others.

### Business Insight

The findings suggest that experience requirements vary across industries in Saudi Arabia.

Some industries tend to seek more experienced candidates, while others are more open to entry-level applicants. This information can help job seekers identify industries that match their experience level and assist employers and policymakers in understanding workforce demand across sectors.

# Statistical Analysis Summary

| Business Question | Statistical Test | Result |
|-------------------|------------------|--------|
| Is there an association between industry and gender requirements? | Chi-Square Test | Significant association (Strong) |
| Is there an association between region and industry? | Chi-Square Test | Significant association (Weak) |
| Do different industries require different average years of experience? | One-Way ANOVA | Significant difference in average experience |